File: front-end-web/src/app/model/MemberDetails.ts

import { Dept } from "./Dept";
import { DeptMember } from "./DeptMember";
import { Member } from "./Member";
import { Position } from "./Position";
import { Role } from "./Role";

export interface MemberDetails {

   member: Member
   allDepts: Dept[]
   allPositions: Position[]
   memberDepts: DeptMember[]
   memberPositions: Role[]

}

File: web-server/src/main/java/com/churchofphilippi/webserver/model/customModels/MemberDetails.java
package com.churchofphilippi.webserver.model.customModels;

import com.churchofphilippi.webserver.model.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MemberDetails {

    private Member member;
    private List<Dept> allDepts;
    private List<Position> allPositions;
    private List<DeptMember> memberDepts;
    private List<Role> memberPositions;

}
